src/modules/flow/flow.c

```diff
@@ -176,18 +176,22 @@ static inline uint32_t compute_hessian_4x6(uint8_t *image, uint16_t x, uint16_t
  * @brief Compute the average pixel gradient of all horizontal and vertical steps
  *
  * TODO compute_diff is not appropriate for low-light mode images
+ * TODO calculate for 8x8 pattern instead of 4x4 to raise number of patches used for flow computation
  *
- * @param image ...
- * @param offX x coordinate of upper left corner of 8x8 pattern in image
- * @param offY y coordinate of upper left corner of 8x8 pattern in image
+ * @param image  the array holding pixel data
+ * @param offX   x coordinate of upper left corner of 8x8 pattern in image
+ * @param offY   y coordinate of upper left corner of 8x8 pattern in image
+ *
+ * @return       value indicating how suitable a pattern is for optical flow calculation
  */
 static inline uint32_t compute_diff(uint8_t *image, uint16_t offX, uint16_t offY, uint16_t row_size)
 {
 	/* calculate position in image buffer */
-	uint16_t off = (offY + 2) * row_size + (offX + 2); // we calc only the 4x4 pattern
+	uint16_t off = (offY + 2) * row_size + (offX + 2);
+    /* we calculate only the center 4x4 pattern */
 	uint32_t acc;
 
-	/* calc row diff */
+	/* calculate SAD row-wise */
```

Comment:
SAD is very generic, I would change it to "calculate vertical gradient"

Reply:
Well, technically it is not gradient, it is sum of absolute values of gradient...

Reply:
The sum is the preparation step to average the gradient - which is not done, what I'm trying to point in the above comment.